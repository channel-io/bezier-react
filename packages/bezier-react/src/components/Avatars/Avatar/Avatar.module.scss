@layer components {
  .Avatar {
    position: relative;
    display: block;
    flex: none;

    &:where(.disabled) {
      pointer-events: none;
      opacity: var(--opacity-disabled);
    }

    &:where(.size-20) {
      width: 20px;
      height: 20px;
    }

    &:where(.size-24) {
      width: 24px;
      height: 24px;
    }

    &:where(.size-30) {
      width: 30px;
      height: 30px;
    }

    &:where(.size-36) {
      width: 36px;
      height: 36px;
    }

    &:where(.size-42) {
      width: 42px;
      height: 42px;
    }

    &:where(.size-48) {
      width: 48px;
      height: 48px;
    }

    &:where(.size-72) {
      width: 72px;
      height: 72px;
    }

    &:where(.size-90) {
      width: 90px;
      height: 90px;
    }

    &:where(.size-120) {
      width: 120px;
      height: 120px;
    }
  }

  .AvatarImage {
    --b-avatar-status-gap: -2px;
    --b-avatar-computed-status-gap: var(--b-avatar-status-gap);

    position: relative;
    box-sizing: content-box;
    display: flex;
    flex-shrink: 0;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    outline: none;
    
    &:where(.big-size) {
      --b-avatar-status-gap: 4px;
    }

    &:where(.bordered[data-state="enabled"]) {
      --b-avatar-computed-status-gap: calc(var(--b-avatar-status-gap) + (2 * var(--b-alpha-smooth-corners-box-shadow-spread-radius)));
    }
  }

  .StatusWrapper {
    position: absolute;
    right: var(--b-avatar-computed-status-gap);
    bottom: var(--b-avatar-computed-status-gap);
    display: flex;
  }
}
