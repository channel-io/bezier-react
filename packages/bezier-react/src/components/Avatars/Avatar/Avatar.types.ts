import type {
  AlphaBezierComponentProps,
  ChildrenProps,
  DisableProps,
  SizeProps,
} from '~/src/types/ComponentProps'

import { type StatusType } from '~/src/components/Status'

export enum AvatarSize {
  Size20 = 20,
  Size24 = 24,
  Size30 = 30,
  Size36 = 36,
  Size42 = 42,
  Size48 = 48,
  Size72 = 72,
  Size90 = 90,
  Size120 = 120,
}

interface AvatarOwnProps {
  /**
   * Semantic name of the avatar.
   */
  name: string
  /**
   * Asset image URL of the avatar.
   * @default ''
   */
  avatarUrl?: string
  /**
   * Fallback image URL for the avatar.
   * @default DEFAULT_AVATAR_URL
   */
  fallbackUrl?: string
  /**
   * Additional status type for this avatar.
   * Typically, this is used to indicate the status of the user (online, offline, etc).
   *
   * If additional customization of the status component is needed,
   * pass the custom status component as `children` of the Avatar component.
   */
  status?: StatusType
  /**
   * Whether to display the border of the avatar.
   * @default false
   */
  showBorder?: boolean
  /**
   * Whether to use the smooth corners feature.
   * Overrides whether the smooth corners feature is applied when it is enabled.
   * @default true
   */
  smoothCorners?: boolean
}

export default interface AvatarProps extends
  AlphaBezierComponentProps,
  React.HTMLAttributes<HTMLDivElement>,
  SizeProps<AvatarSize>,
  DisableProps,
  ChildrenProps,
  AvatarOwnProps {}
