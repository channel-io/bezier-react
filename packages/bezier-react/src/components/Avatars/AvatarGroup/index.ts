import { AvatarGroup } from './AvatarGroup'
import { AvatarGroupEllipsisType } from './AvatarGroup.types'
import type AvatarGroupProps from './AvatarGroup.types'

export type {
  AvatarGroupProps,
}

export {
  AvatarGroup,
  AvatarGroupEllipsisType,
}
