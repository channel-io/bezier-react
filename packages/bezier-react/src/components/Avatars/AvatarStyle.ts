export const AVATAR_BORDER_RADIUS_PERCENTAGE = 42

export const AVATAR_BORDER_WIDTH = 2

export const AVATAR_GROUP_DEFAULT_SPACING = 4
