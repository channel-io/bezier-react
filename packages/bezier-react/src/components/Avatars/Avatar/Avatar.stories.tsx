/* External dependencies */
import React from 'react'
import { base } from 'paths.macro'
import { Story, Meta } from '@storybook/react'

/* Internal dependencies */
import { styled } from 'Foundation'
import { getTitle } from 'Utils/storyUtils'
import { StatusType } from 'Components/Status'
import { Avatar } from './Avatar'
import AvatarProps, { AvatarSize } from './Avatar.types'

const MOCK_AVATAR_URL = 'https://cf.channel.io/thumb/200x200/pub-file/1/606d87d059a6093594c0/ch-symbol-filled-smiley-bg.png'

const avatarSizeList = Object.keys(AvatarSize)
  .filter(value => Number.isNaN(Number(value)) === true)
  .map(key => AvatarSize[key])

const statusTypeList = Object.keys(StatusType)
  .map(key => StatusType[key])

export default {
  title: getTitle(base),
  component: Avatar,
  argTypes: {
    size: {
      control: {
        type: 'radio',
        options: avatarSizeList,
      },
    },
    status: {
      control: {
        type: 'radio',
        options: [
          undefined,
          ...statusTypeList,
        ],
      },
    },
    onClick: {
      action: 'clicked',
    },
    onMouseEnter: {
      action: 'mouseEnter',
    },
    onMouseLeave: {
      action: 'mouseLeave',
    },
  },
} as Meta

// NOTE: (@ed) border 색상을 명확하게 보여주기 위해 회색의 Wrapper를 추가했습니다
const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 200px;
  height: 200px;
  background-color: ${({ foundation }) => foundation?.theme?.['bg-grey-light']};
`

const Template: Story<AvatarProps> = (args) => (
  <Wrapper>
    <Avatar {...args} />
  </Wrapper>
)

export const Primary: Story<AvatarProps> = Template.bind({})
Primary.args = {
  avatarUrl: MOCK_AVATAR_URL,
  name: 'Channel',
  size: AvatarSize.Size24,
  showBorder: false,
  disabled: false,
}

const TemplateWithCustomStatus: Story<AvatarProps> = (args) => (
  <Wrapper>
    <Avatar {...args}>
      <Avatar
        avatarUrl="https://bit.ly/kent-c-dodds"
        name="Kent Dodds"
        size={AvatarSize.Size20}
        showBorder
      />
    </Avatar>
  </Wrapper>
)

export const WithCustomStatus: Story<AvatarProps> = TemplateWithCustomStatus.bind({})
WithCustomStatus.args = {
  avatarUrl: MOCK_AVATAR_URL,
  name: 'Channel',
  size: AvatarSize.Size48,
  showBorder: false,
  disabled: false,
}
