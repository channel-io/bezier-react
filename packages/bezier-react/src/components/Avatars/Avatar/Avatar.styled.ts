/* Internal denpendencies */
import { styled, css, smoothCorners } from 'Foundation'
import { enableSmoothCorners } from 'Worklets/EnableCSSHoudini'
import type { InterpolationProps } from 'Types/Foundation'
import DisabledOpacity from 'Constants/DisabledOpacity'
import { AVATAR_BORDER_WIDTH, AVATAR_BORDER_RADIUS_PERCENTAGE } from 'Components/Avatars/AvatarStyle'
import { AvatarSize } from './Avatar.types'

interface AvatarWrapperProps extends InterpolationProps {
  disabled: boolean
}

interface AvatarProps extends InterpolationProps {
  size: AvatarSize
  showBorder: boolean
}

interface StatusWrapperProps extends Pick<AvatarProps, 'showBorder' | 'size'> {}

function calcStatusGap({ showBorder, size }: StatusWrapperProps) {
  let gap = (size >= AvatarSize.Size72 ? 4 : -2)
  if (showBorder && enableSmoothCorners.current) {
    gap += AVATAR_BORDER_WIDTH * 2
  }
  return gap
}

const disabledStyle = css`
  opacity: ${DisabledOpacity};
`

const smoothCornersFallbackBorderStyle = css`
  &::after {
    ${({ foundation }) => foundation?.border?.getBorder(AVATAR_BORDER_WIDTH, foundation?.theme?.['bg-white-high'])};

    position: absolute;
    top: -${AVATAR_BORDER_WIDTH}px;
    left: -${AVATAR_BORDER_WIDTH}px;
    z-index: -1;
    box-sizing: content-box;
    display: block;
    width: 100%;
    height: 100%;
    content: '';
    background-color: ${({ foundation }) => `${foundation?.theme?.['bg-white-normal']}`};
    border-radius: ${AVATAR_BORDER_RADIUS_PERCENTAGE}%;
  }
`

export const AvatarImage = styled.div<AvatarProps>`
  position: relative;
  box-sizing: content-box;
  display: flex;
  flex-shrink: 0;
  align-items: center;
  justify-content: center;
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;
  outline: none;

  ${({ showBorder }) => (!enableSmoothCorners.current && showBorder) && smoothCornersFallbackBorderStyle}

  ${({ foundation, showBorder }) => smoothCorners({
    shadow: showBorder ? `0 0 0 ${AVATAR_BORDER_WIDTH}px ${foundation?.theme?.['bg-white-high']}` : undefined,
    shadowBlur: showBorder ? AVATAR_BORDER_WIDTH : 0,
    backgroundColor: foundation?.theme?.['bg-white-normal'],
    borderRadius: `${AVATAR_BORDER_RADIUS_PERCENTAGE}%`,
    hasBackgroundImage: true,
  })};

  ${({ interpolation }) => interpolation}
`

export const AvatarImageWrapper = styled.div`
  position: relative;
  z-index: 1;
`

export const AvatarWrapper = styled.div<AvatarWrapperProps>`
  position: relative;
  z-index: 0;

  ${({ disabled }) => disabled && disabledStyle}

  ${({ interpolation }) => interpolation}
`

export const StatusWrapper = styled.div<StatusWrapperProps>`
  position: absolute;
  right: ${calcStatusGap}px;
  bottom: ${calcStatusGap}px;
`
