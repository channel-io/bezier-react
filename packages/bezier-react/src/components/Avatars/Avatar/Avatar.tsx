import React, {
  forwardRef,
  useMemo,
} from 'react'

import classNames from 'classnames'

import { isEmpty } from '~/src/utils/typeUtils'

import { type BoxShadow } from '~/src/components/AlphaSmoothCornersBox'
import { AVATAR_BORDER_RADIUS_PERCENTAGE } from '~/src/components/Avatars/AvatarStyle'
import {
  Status,
  StatusSize,
} from '~/src/components/Status'

// NOTE: Don't fix it. When using absolute paths, a module resolve error occurs at the runtime of the library consumer.
// eslint-disable-next-line no-restricted-imports
import defaultAvatarUrl from '../assets/defaultAvatar.svg'

import type AvatarProps from './Avatar.types'
import { AvatarSize } from './Avatar.types'
import useProgressiveImage from './useProgressiveImage'

import * as Styled from './Avatar.styled'

const shadow: BoxShadow = {
  spreadRadius: 2,
  color: 'bg-white-high',
}

export const AVATAR_WRAPPER_TEST_ID = 'bezier-react-avatar-wrapper'
export const AVATAR_TEST_ID = 'bezier-react-avatar'
export const STATUS_WRAPPER_TEST_ID = 'bezier-react-status-wrapper'

export const Avatar = forwardRef(function Avatar({
  avatarUrl = '',
  fallbackUrl = defaultAvatarUrl,
  size = AvatarSize.Size24,
  name,
  testId = AVATAR_TEST_ID,
  disabled = false,
  showBorder = false,
  smoothCorners = true,
  status,
  className: classNameProp,
  wrapperClassName,
  interpolation,
  wrapperInterpolation,
  style: styleProp,
  children,
  ...rest
}: AvatarProps, forwardedRef: React.Ref<HTMLDivElement>) {
  const loadedAvatarUrl = useProgressiveImage(avatarUrl, fallbackUrl)

  const StatusComponent = useMemo(() => {
    if (
      (isEmpty(children) && !status)
      || (children && !React.isValidElement(children))
    ) {
      return null
    }

    const statusSize = size >= AvatarSize.Size90 ? StatusSize.L : StatusSize.M

    const Contents = (() => {
      if (children) { return children }
      if (status) {
        return (
          <Status
            type={status}
            size={statusSize}
          />
        )
      }
      return null
    })()

    return (
      <Styled.StatusWrapper data-testid={STATUS_WRAPPER_TEST_ID}>
        { Contents }
      </Styled.StatusWrapper>
    )
  }, [
    status,
    size,
    children,
  ])

  const style = useMemo(() => ({
    ...styleProp,
    '--bezier-avatar-status-gap': `${size >= AvatarSize.Size72 ? 4 : -2}px`,
  }), [
    styleProp,
    size,
  ])

  const className = classNames(
    classNameProp,
    size && `size-${size}`,
    showBorder && 'bordered',
  )

  return (
    <Styled.AvatarWrapper
      data-testid={AVATAR_WRAPPER_TEST_ID}
      data-disabled={disabled}
      className={wrapperClassName}
      interpolation={wrapperInterpolation}
    >
      <Styled.AvatarImage
        {...rest}
        ref={forwardedRef}
        data-testid={testId}
        aria-label={name}
        style={style}
        className={className}
        interpolation={interpolation}
        disabled={!smoothCorners}
        borderRadius={`${AVATAR_BORDER_RADIUS_PERCENTAGE}%`}
        shadow={showBorder ? shadow : undefined}
        backgroundColor="bg-white-normal"
        backgroundImage={loadedAvatarUrl}
      >
        { StatusComponent }
      </Styled.AvatarImage>
    </Styled.AvatarWrapper>
  )
})
