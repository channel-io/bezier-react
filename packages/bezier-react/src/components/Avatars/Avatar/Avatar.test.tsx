import React from 'react'

import DisabledOpacity from '~/src/constants/DisabledOpacity'
import { render } from '~/src/utils/testUtils'

import { AVATAR_BORDER_RADIUS_PERCENTAGE } from '~/src/components/Avatars/AvatarStyle'
import { StatusType } from '~/src/components/Status'

import {
  AVATAR_TEST_ID,
  AVATAR_WRAPPER_TEST_ID,
  Avatar,
  STATUS_WRAPPER_TEST_ID,
} from './Avatar'
import type AvatarProps from './Avatar.types'
import { AvatarSize } from './Avatar.types'

jest.mock('~/src/worklets/EnableCSSHoudini', () => ({
  __esModule: true,
  ...jest.requireActual('~/src/worklets/EnableCSSHoudini') as object,
  enableSmoothCorners: { current: true },
}))

describe('Avatar >', () => {
  let props: AvatarProps

  const mockAvatarUrl = 'https://bit.ly/dan-abramov'
  const mockFallbackUrl = 'https://www.google.com'

  beforeEach(() => {
    props = {
      avatarUrl: mockAvatarUrl,
      fallbackUrl: mockFallbackUrl,
      name: 'Name',
    }
  })

  afterAll(() => {
    jest.restoreAllMocks()
  })

  // NOTE: unavailable smoothCorners
  const renderAvatar = (optionProps?: Partial<AvatarProps>) => render(
    <Avatar {...props} {...optionProps} />,
  )

  it('Snapshot', () => {
    const { getByTestId } = renderAvatar()
    const img = getByTestId(AVATAR_TEST_ID)

    expect(img).toMatchSnapshot()
  })

  it('renders image with correct style', () => {
    const { getByTestId } = renderAvatar()
    const img = getByTestId(AVATAR_TEST_ID)

    expect(img).toHaveStyle('position: relative')
    expect(img).toHaveStyle('box-sizing: content-box')
    expect(img).toHaveStyle('display: flex')
    expect(img).toHaveStyle('outline: none')
    expect(img).toHaveStyle('background-image: var(--background-image)')
    expect(img).toHaveStyle(`--background-image: url(${mockFallbackUrl})`)
    expect(img).toHaveStyle('background-size: cover')
    expect(img).toHaveStyle(`border-radius: ${AVATAR_BORDER_RADIUS_PERCENTAGE}%`)
  })

  it('renders disabled style', () => {
    const { getByTestId } = renderAvatar({ disabled: true })
    const wrapper = getByTestId(AVATAR_WRAPPER_TEST_ID)

    expect(wrapper).toHaveStyle(`opacity: ${DisabledOpacity}`)
  })

  it('renders border style', () => {
    const { getByTestId } = renderAvatar({ showBorder: true })
    const img = getByTestId(AVATAR_TEST_ID)

    // NOTE(@ed): psuedo element 스타일 테스트가 어려워, 스냅샷 테스트로 대체
    expect(img).toMatchSnapshot()
  })

  it('should have right -2px, bottom -2px style on StatusWrapper', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 2px, bottom 2px style on StatusWrapper when show border', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, showBorder: true })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 4px, bottom 4px style on StatusWrapper when size grater then 72', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size72 })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 8px, bottom 8px style on StatusWrapper when size grater then 72 and show border', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size72, showBorder: true })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 4px, bottom 4px style on StatusWrapper when size grater then 90', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size90 })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 8px, bottom 8px style on StatusWrapper when size grater then 90 and show border', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size90, showBorder: true })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })
})
