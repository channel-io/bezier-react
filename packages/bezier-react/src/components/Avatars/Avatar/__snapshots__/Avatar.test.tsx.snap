// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar > Snapshot 1`] = `
.c0 {
  --b-alpha-smooth-corners-box-border-radius: 0;
  --b-alpha-smooth-corners-box-shadow-offset-x: 0;
  --b-alpha-smooth-corners-box-shadow-offset-y: 0;
  --b-alpha-smooth-corners-box-shadow-blur-radius: 0px;
  --b-alpha-smooth-corners-box-shadow-spread-radius: 0px;
  --b-alpha-smooth-corners-box-shadow-color: transparent;
  --b-alpha-smooth-corners-box-padding: 0px;
  --b-alpha-smooth-corners-box-margin: 0px;
  --b-alpha-smooth-corners-box-background-color: transparent;
  box-sizing: content-box;
  margin: var(--b-alpha-smooth-corners-box-margin);
  background-color: var(--b-alpha-smooth-corners-box-background-color);
  background-image: var(--b-alpha-smooth-corners-box-background-image);
  background-size: cover;
  border-radius: var(--b-alpha-smooth-corners-box-border-radius);
  box-shadow: var(--b-alpha-smooth-corners-box-shadow-offset-x) var(--b-alpha-smooth-corners-box-shadow-offset-y) var(--b-alpha-smooth-corners-box-shadow-blur-radius) var(--b-alpha-smooth-corners-box-shadow-spread-radius) var(--b-alpha-smooth-corners-box-shadow-color);
}

@supports (background:paint(smooth-corners)) {
  .c0[data-state="enabled"] {
    padding: var(--b-alpha-smooth-corners-box-padding);
    margin: calc(var(--b-alpha-smooth-corners-box-margin) + (-1 * var(--b-alpha-smooth-corners-box-padding)));
    background: paint(smooth-corners);
    border-radius: 0;
    border-image-source: var(--b-alpha-smooth-corners-box-background-image);
    box-shadow: none;
    --smooth-corners: var(--b-alpha-smooth-corners-box-border-radius);
    --smooth-corners-shadow: var(--b-alpha-smooth-corners-box-shadow-offset-x) var(--b-alpha-smooth-corners-box-shadow-offset-y) var(--b-alpha-smooth-corners-box-shadow-blur-radius) var(--b-alpha-smooth-corners-box-shadow-spread-radius) var(--b-alpha-smooth-corners-box-shadow-color);
    --smooth-corners-bg-color: var(--b-alpha-smooth-corners-box-background-color);
    --smooth-corners-padding: var(--b-alpha-smooth-corners-box-padding);
    --smooth-corners-radius-unit: var(--b-alpha-smooth-corners-box-border-radius-type);
  }
}

<div
  aria-label="Name"
  class="c0 AvatarImage"
  data-state="disabled"
  data-testid="bezier-react-avatar"
  style="--b-alpha-smooth-corners-box-border-radius: 42%; --b-alpha-smooth-corners-box-border-radius-type: string; --b-alpha-smooth-corners-box-shadow-blur-radius: 0px; --b-alpha-smooth-corners-box-shadow-spread-radius: 0px; --b-alpha-smooth-corners-box-padding: 0px; --b-alpha-smooth-corners-box-margin: 0px; --b-alpha-smooth-corners-box-background-color: var(--bg-white-normal); --b-alpha-smooth-corners-box-background-image: url(https://www.google.com);"
/>
`;

exports[`Avatar > renders border style 1`] = `
.c0 {
  --b-alpha-smooth-corners-box-border-radius: 0;
  --b-alpha-smooth-corners-box-shadow-offset-x: 0;
  --b-alpha-smooth-corners-box-shadow-offset-y: 0;
  --b-alpha-smooth-corners-box-shadow-blur-radius: 0px;
  --b-alpha-smooth-corners-box-shadow-spread-radius: 0px;
  --b-alpha-smooth-corners-box-shadow-color: transparent;
  --b-alpha-smooth-corners-box-padding: 0px;
  --b-alpha-smooth-corners-box-margin: 0px;
  --b-alpha-smooth-corners-box-background-color: transparent;
  box-sizing: content-box;
  margin: var(--b-alpha-smooth-corners-box-margin);
  background-color: var(--b-alpha-smooth-corners-box-background-color);
  background-image: var(--b-alpha-smooth-corners-box-background-image);
  background-size: cover;
  border-radius: var(--b-alpha-smooth-corners-box-border-radius);
  box-shadow: var(--b-alpha-smooth-corners-box-shadow-offset-x) var(--b-alpha-smooth-corners-box-shadow-offset-y) var(--b-alpha-smooth-corners-box-shadow-blur-radius) var(--b-alpha-smooth-corners-box-shadow-spread-radius) var(--b-alpha-smooth-corners-box-shadow-color);
}

@supports (background:paint(smooth-corners)) {
  .c0[data-state="enabled"] {
    padding: var(--b-alpha-smooth-corners-box-padding);
    margin: calc(var(--b-alpha-smooth-corners-box-margin) + (-1 * var(--b-alpha-smooth-corners-box-padding)));
    background: paint(smooth-corners);
    border-radius: 0;
    border-image-source: var(--b-alpha-smooth-corners-box-background-image);
    box-shadow: none;
    --smooth-corners: var(--b-alpha-smooth-corners-box-border-radius);
    --smooth-corners-shadow: var(--b-alpha-smooth-corners-box-shadow-offset-x) var(--b-alpha-smooth-corners-box-shadow-offset-y) var(--b-alpha-smooth-corners-box-shadow-blur-radius) var(--b-alpha-smooth-corners-box-shadow-spread-radius) var(--b-alpha-smooth-corners-box-shadow-color);
    --smooth-corners-bg-color: var(--b-alpha-smooth-corners-box-background-color);
    --smooth-corners-padding: var(--b-alpha-smooth-corners-box-padding);
    --smooth-corners-radius-unit: var(--b-alpha-smooth-corners-box-border-radius-type);
  }
}

<div
  aria-label="Name"
  class="c0 AvatarImage bordered"
  data-state="disabled"
  data-testid="bezier-react-avatar"
  style="--b-alpha-smooth-corners-box-border-radius: 42%; --b-alpha-smooth-corners-box-border-radius-type: string; --b-alpha-smooth-corners-box-shadow-blur-radius: 0px; --b-alpha-smooth-corners-box-shadow-spread-radius: 2px; --b-alpha-smooth-corners-box-shadow-color: var(--bg-white-high); --b-alpha-smooth-corners-box-padding: 4px; --b-alpha-smooth-corners-box-margin: 0px; --b-alpha-smooth-corners-box-background-color: var(--bg-white-normal); --b-alpha-smooth-corners-box-background-image: url(https://www.google.com);"
/>
`;

exports[`Avatar > should have right -2px, bottom -2px style on StatusWrapper 1`] = `
@supports (background:paint(smooth-corners)) {

}

<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m margin"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 2px, bottom 2px style on StatusWrapper when show border 1`] = `
@supports (background:paint(smooth-corners)) {

}

<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m margin"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 4px, bottom 4px style on StatusWrapper when size grater then 72 1`] = `
@supports (background:paint(smooth-corners)) {

}

<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m margin"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 4px, bottom 4px style on StatusWrapper when size grater then 90 1`] = `
@supports (background:paint(smooth-corners)) {

}

<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-l margin"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 8px, bottom 8px style on StatusWrapper when size grater then 72 and show border 1`] = `
@supports (background:paint(smooth-corners)) {

}

<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m margin"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 8px, bottom 8px style on StatusWrapper when size grater then 90 and show border 1`] = `
@supports (background:paint(smooth-corners)) {

}

<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-l margin"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;
